api/v1alpha1/driver_types.go

```diff
@@ -0,0 +1,271 @@
+/*
+Copyright 2024.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	storagev1 "k8s.io/api/storage/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type LogSpec struct {
+	// Log level for driver pods,
+	// Supported values from 0 to 5. 0 for general useful logs (the default), 5 for trace level verbosity.
+	// Default to 0
+	LogLevel   int `json:"logLevel,omitempty"`
+	MaxFiles   int `json:"maxFiles,omitempty"`
+	MaxLogSize int `json:"maxLogSize,omitempty"`
+}
+
+// TODO: Add a porper description
+type SnapshotPolicyType string
+
+const (
+	AutoDetectSnapshotPolicy SnapshotPolicyType = "auto-detect"
+	GroupSnapshotPolicy      SnapshotPolicyType = "group"
+	SnapshotSnapshotPolicy   SnapshotPolicyType = "snapshot"
+)
+
+type EncryptionSpec struct {
+	ConfigMapRef corev1.LocalObjectReference `json:"configMapName,omitempty"`
+}
+
+type PodCommonSpec struct {
+	// Pod's user defined priority class name
+	PrioritylClassName *string `json:"priorityClassName,omitempty"`
+
+	// Pod's labels
+	Labels map[string]string `json:"labels,omitempty"`
+
+	// Pod's annotations
+	Annotations map[string]string `json:"annotations,omitempty"`
+
+	// Pod's affinity settings
+	Affinity *corev1.Affinity `json:"affinity,omitempty"`
+
+	// Pod's tolerations list
+	Tolerations []corev1.Toleration `json:"tolerations,omitempty"`
+}
+
+type PluginResourcesSpec struct {
+	Registrar *corev1.ResourceRequirements `json:"registerer,omitempty"`
+	Liveness  *corev1.ResourceRequirements `json:"liveness,omitempty"`
+	Plugin    *corev1.ResourceRequirements `json:"plugin,omitempty"`
+}
+
+type PluginSpec struct {
+	// Embedded common pods spec
+	PodCommonSpec `json:"inline"`
+
+	// Driver's plugin daemonset update strategy, supported values are OnDelete and RollingUpdate.
+	// Default value is RollingUpdate with MaxAvailabile set to 1
+	UpdateStrategy *appsv1.DaemonSetUpdateStrategy `json:"updateStrategy,omitempty"`
+
+	// Resource requirements for plugin's containers
+	Resources PluginResourcesSpec `json:"resources,omitempty"`
+
+	Volumes []corev1.Volume `json:"pluginVolumes,omitempty"`
```

Comment:
`pluginVolumes` to `volumes` as its already part of Plugin?

Reply:
👍

Reply:
done